packages/liveblocks-node/src/client.ts

```diff
@@ -1612,6 +1651,59 @@ export class Liveblocks {
     return convertToThreadData((await res.json()) as ThreadDataPlain<M>);
   }
 
+  /**
+   * Subscribes a user to a thread.
+   * @param params.roomId The room ID of the thread.
+   * @param params.threadId The thread ID to subscribe to.
+   * @param params.data.userId The user ID of the user to subscribe to the thread.
+   * @param options.signal (optional) An abort signal to cancel the request.
+   * @returns The thread subscription.
+   */
+  public async subscribeToThread(
+    params: { roomId: string; threadId: string; data: { userId: string } },
+    options?: RequestOptions
+  ): Promise<SubscriptionData> {
+    const { roomId, threadId } = params;
+
+    const res = await this.#post(
+      url`/v2/rooms/${roomId}/threads/${threadId}/subscribe`,
+      { userId: params.data.userId },
+      options
+    );
+
+    if (!res.ok) {
+      throw await LiveblocksError.from(res);
+    }
+
+    return convertToSubscriptionData(
+      (await res.json()) as SubscriptionDataPlain
+    );
+  }
+
+  /**
+   * Unsubscribes a user from a thread.
+   * @param params.roomId The room ID of the thread.
+   * @param params.threadId The thread ID to unsubscribe from.
+   * @param params.data.userId The user ID of the user to unsubscribe from the thread.
+   * @param options.signal (optional) An abort signal to cancel the request.
+   */
+  public async unsubscribeFromThread(
+    params: { roomId: string; threadId: string; data: { userId: string } },
+    options?: RequestOptions
+  ): Promise<void> {
+    const { roomId, threadId } = params;
+
+    const res = await this.#post(
+      url`/v2/rooms/${roomId}/threads/${threadId}/unsubscribe`,
+      { userId: params.data.userId },
+      options
+    );
+
+    if (!res.ok) {
+      throw await LiveblocksError.from(res);
+    }
+  }
```

Comment:
It's a good practice to call `res.json()` even if you don't need to return anything in this `void` function.

The reason is that `res.ok` only checks the status header, and it does not start consuming the response body. This might cause the server to hold on to the response body stream longer than needed, because it hasn't been consumed by the client yet.

(I haven't checked the backend implementation, but if you haven't defined a response body in the backend for this endpoint, it might be good to at the very least return a minimal `{"ok":true}` JSON body response there. Ideally, _all_ endpoint responses are JSON.)

Reply:
Yeah, ideally we stop using empty 204 responses entirely in the backend. It makes it a backward incompatible change if we need to change it. I have been through that a couple times and think at this point that every endpoint should return a JSON object in the body response, no exception. It just makes it a lot easier to extend later. cc @FlowFlorent

Reply:
Traditionally our DELETEs do not return the resource so that's why it feels ok to use 204. But honestly returning a 200 so that we can extend it later sounds like a lot easier time, on board with that!

Reply:
Are we okay with the inconsistency between this and all our other 204s? If so I'll update the endpoint and add `await res.json()` here.